tests/foreman/api/test_errata.py

```diff
@@ -1384,15 +1382,18 @@ def test_errata_installation_with_swidtags(
 
     :steps:
 
-        1. create product and repository having swid tags
-        2. create content view and published it with repository
-        3. create activation key and register content host
-        4. create rhel8, swid repos on content host
-        5. install swid-tools, dnf-plugin-swidtags packages on content host
-        6. install older module stream and generate errata, swid tag
-        7. assert errata count, swid tags are generated
-        8. install errata vis updating module stream
-        9. assert errata count and swid tag after module update
+        1. promote empty content view and create activation key
```

Comment:
Is this step really necessary? I mean for AK creation you just need a CV, not CVV.

Reply:
@lhellebr @vsedmik 
It works in webUI, but the api is still giving trouble. 
Please try this on your local setup, for me fails in stream, 6.15, and 6.14 latest:
```
# read org fixture and create LCE
_org = module_sca_manifest_org.read()
_lce = target_sat.api.LifecycleEnvironment(
    organization=module_sca_manifest_org
).create()

# create empty CV
_cv = target_sat.api.ContentView(
    organization=_org,
    environment=[_lce],
).create()

# create AK associated to LCE and CV
_ak = target_sat.api.ActivationKey(
    organization=_org,
    environment=_lce,
    content_view=_cv,
).create()
```
Unless I publish the creating the new CV (such as with local method `cv_publish_promote`), 
I will get 422 upon creating the activationkey:
```
requests.exceptions.HTTPError: 422 Client Error: Unprocessable Content for url: https:/hostname/katello/api/v2/activation_keys
```

Reply:
I tried the way suggested by @vsedmik , but it doesn't work for me
to create AK, I need to `publish cv > read cv version > Promote cvv by specifing LCE name > create ak`
@lhellebr

Reply:
Perhaps the difference is that you use Library in WebUI? The error when I try through API is:
```
Validation failed: Content view 'WZhNaQdK' is not in environment 'RKwDCEqT' (ActiveRecord::RecordInvalid)
```
This works for me, I'll leave it up to you if it suits your testcase.
```
import pytest
@pytest.mark.rhel_ver_match('8')
def test_mintest(
    module_target_sat, function_sca_manifest_org
):
    org = function_sca_manifest_org
    #lce = module_target_sat.api.LifecycleEnvironment(organization=org).create()
    content_view = module_target_sat.api.ContentView(organization=org).create()
    #content_view.publish()
    #content_view = content_view.read()
    #content_view.version[-1].promote(data={'environment_ids': lce.id})
    activation_key = module_target_sat.api.ActivationKey(
        content_view=content_view,
        organization=org,
        environment=module_target_sat.api.LifecycleEnvironment(name='Library'),
    ).create()
```

Reply:
Thank you for addressing this, marking as resolved.